# Part 1

### TODO 1.1

In [28]:
from sklearn.datasets import fetch_20newsgroups
from collections import Counter

train_data = fetch_20newsgroups(subset='train')
test_data = fetch_20newsgroups(subset='test')

y_train = train_data.target
target_names = train_data.target_names
counts_train = Counter(y_train)

print(f"{'Category':<30}   {'Count'}")
for i in range(len(target_names)):
    print(f"{target_names[i]:<30} | {counts_train[i]}")

majority_baseline_accuracy = max(counts_train.values()) / len(y_train)
print(f"\nMajority Baseline Train Accuracy: {majority_baseline_accuracy:.5f}")

y_test = test_data.target
counts_test = Counter(y_test)

majority_baseline_test_accuracy = max(counts_test.values()) / len(y_test)
print(f"Majority Baseline Test Accuracy: {majority_baseline_test_accuracy:.5f}")


Category                         Count
alt.atheism                    | 480
comp.graphics                  | 584
comp.os.ms-windows.misc        | 591
comp.sys.ibm.pc.hardware       | 590
comp.sys.mac.hardware          | 578
comp.windows.x                 | 593
misc.forsale                   | 585
rec.autos                      | 594
rec.motorcycles                | 598
rec.sport.baseball             | 597
rec.sport.hockey               | 600
sci.crypt                      | 595
sci.electronics                | 591
sci.med                        | 594
sci.space                      | 593
soc.religion.christian         | 599
talk.politics.guns             | 546
talk.politics.mideast          | 564
talk.politics.misc             | 465
talk.religion.misc             | 377

Majority Baseline Train Accuracy: 0.05303
Majority Baseline Test Accuracy: 0.05297


The majority baseline on the training set is 0.05303 and the majority baseline on the test set is 0.05297.

### TODO 1.2